src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentMappingService.cs

```diff
@@ -26,6 +27,8 @@ internal abstract class RazorDocumentMappingService
 
         public abstract bool TryMapToProjectedDocumentOrNextCSharpPosition(RazorCodeDocument codeDocument, int absoluteIndex, [NotNullWhen(true)] out Position? projectedPosition, out int projectedIndex);
 
+        public abstract bool TryMapToVSProjectedDocumentPosition(RazorCodeDocument codeDocument, int absoluteIndex, [NotNullWhen(true)] out VSPosition? projectedPosition, out int projectedIndex);
```

Comment:
As per ryan's comment, these things make more sense as extension methods to me

Reply:
Rebased to use Ryan's variant.